### PR TITLE
[registry-facade] Remove feature flag

### LINKED ISSUE #2655: Make registry-facade standard
We have had registry-facade in our code base for almost a year now, and still it is only a variant. We need to reduce the variance within Gitpod, hence want to move towards a registry-facade only setup.

To this end, we need to:
- remove the registry-facade variance at deploy time
- remove the registry-facade feature flag and all variant code
- remove the static Theia server

Related issues:
- support zero-downtime handover (#2512)
- make HTTPS deployment the default in dev-staging (#2704)
- asses the impact on self-hosted and update the docs accordingly. I'd expect this to possibly make some ingress mode variants obsolete. (#2683)

### PR DESCRIPTION
**Pull request by csweichel**
This PR removes all variance surrounding registry-facade. Once this is merged, registry-facade is the sole way to start workspaces. To this end, we remove
- the `registry_facade` feature flag and all variance it created in ws-manager
- node-daemon entirely. The remaining few `sysctls` are now an init container on ws-daemon
- Theia's static-server, because now that registry-facade is standard, so is blobserve.

### How to test
1. Start a workspace - if it starts, all is well

fixes #2655

### CODE REVIEW

File: components/ws-proxy/pkg/proxy/routes.go [hunk 1]
```diff
@@ -188,11 +188,6 @@ func (ir *ideRoutes) HandleSupervisorFrontendRoute(route *mux.Route) {
 }
 
 func (ir *ideRoutes) HandleRoot(route *mux.Route) {
-	if ir.Config.Config.BlobServer == nil {
-		ir.handleRootWithoutBlobserve(route)
-		return
-	}
```
Comment (corneliusludmann) [inline components/ws-proxy/pkg/proxy/routes.go:194]: [With the change in the `ws-proxy-configmap.yaml` above](https://github.com/gitpod-io/gitpod/pull/3182/files#diff-e94a32e97df84e397163a88899dd32feaf13585c4bafac5eb022a7e526de1ca0R65) it's still possible that the `blobServer` config is missing (when `blobserve` component is disabled). Doesn't that conflict with removing this case here?

**Comment by corneliusludmann** [comment]
I have also found a few lines with references to node-daemon. :laughing: 

https://github.com/gitpod-io/gitpod/blob/ff9d321050c631131cf094705a5d242eb31b21f4/chart/templates/ws-manager-configmap.yaml#L81

This one breaks `helm` run when not using `values.dev.yaml`. This is because `values.dev.yaml` has:

https://github.com/gitpod-io/gitpod/blob/ff9d321050c631131cf094705a5d242eb31b21f4/.werft/values.dev.yaml#L114-L116


And here are two self-hosted files with references to node-daemon. Don't know if you want to address it in this PR as well or create an issue for this:

https://github.com/gitpod-io/gitpod/blob/ff9d321050c631131cf094705a5d242eb31b21f4/docs/self-hosted/install/nodes.md#L29-L33

https://github.com/gitpod-io/gitpod/blob/ff9d321050c631131cf094705a5d242eb31b21f4/install/gcp-terraform/modules/gitpod/templates/values.node-layout.tpl#L5-L9

**Comment by csweichel** [comment]
Thank you for pointing those bits out. I've gone over the PR one more time and removed these bits.

**Review by corneliusludmann — APPROVED** [review]
Tested with the preview environment and self-hosted (k3s image). Both work fine.

**Pull request merged**